lib/sycl/SYCL.jl

```diff
@@ -73,6 +73,7 @@ mutable struct syclQueue
     syclQueueCreate(handle, ctx, ze_queue, true)
     obj = new(handle[], ctx, ze_queue)
     finalizer(obj) do queue
+      onemklDestroy()
```

Comment:
I don't think `onemklDestroy` (cfr. `cublasDestroy`) is a correct name here, because we're not really destroying a library handle, just the cache which (surprisingly) retains the SYCL handles that were passed as an argument to `gemm` and others.

Reply:
I was expecting that oneMKL will add a Destroy() function similar to cublas, hence the name.  How about onemklClose() or onemklCleanup()?

cublas defines a handle object that is initialized and destroyed because CUDA has implicit current stream, which needs to be recorded for subsequent calls to cublas.  In SYCL, the queue is always passed as an explicit object, so oneMKL team probably thought that it doesn't need to define an additional handle object.  However,  as we try to fix here, oneMKL also needs an explicit function similar to cublasDestroy to release the cached objects.